soh/src/code/z_map_exp.c

```diff
@@ -855,58 +858,50 @@ void Minimap_Draw(PlayState* play) {
+
+                        s16 entranceX = topLeft0 ? OTRGetRectDimensionFromLeftEdge(PosX) : OTRGetRectDimensionFromRightEdge(PosX);
+                        s16 entranceY = newY + (topLeft0 ? 0 : Y_Margins_Minimap);
                         if (CVarGetInteger("gMinimapPosType", 0) != 0) {
-                            PosX = gMapData->owEntranceIconPosX[sEntranceIconMapIndex] + CVarGetInteger("gMinimapPosX", 0) + X_Margins_Minimap;
-                            PosY = gMapData->owEntranceIconPosY[sEntranceIconMapIndex] + CVarGetInteger("gMinimapPosY", 0) + Y_Margins_Minimap;
-                            if (CVarGetInteger("gMinimapPosType", 0) == 1) {//Anchor Left
-                                if (CVarGetInteger("gMinimapUseMargins", 0) != 0) {X_Margins_Minimap = Left_MM_Margin;};
-                                TopLeftX = OTRGetRectDimensionFromLeftEdge(PosX) << 2;
-                                TopLeftW = OTRGetRectDimensionFromLeftEdge(PosX + IconSize) << 2;
-                            } else if (CVarGetInteger("gMinimapPosType", 0) == 2) {//Anchor Right
-                                if (CVarGetInteger("gMinimapUseMargins", 0) != 0) {X_Margins_Minimap = Right_MM_Margin;};
-                                TopLeftX = OTRGetRectDimensionFromRightEdge(PosX) << 2;
-                                TopLeftW = OTRGetRectDimensionFromRightEdge(PosX + IconSize) << 2;
-                            } else if (CVarGetInteger("gMinimapPosType", 0) == 3) {//Anchor None
-                                TopLeftX = PosX << 2;
-                                TopLeftW = PosX + IconSize << 2;
+                            if (!topLeft0) {
+                                entranceY = newY + CVarGetInteger("gMinimapPosY", 0) + Y_Margins_Minimap;
+                                if (CVarGetInteger("gMinimapPosType", 0) == 1) {//Anchor Left
+                                    if (CVarGetInteger("gMinimapUseMargins", 0) != 0) {X_Margins_Minimap = Left_MM_Margin;};
+                                    entranceX = OTRGetRectDimensionFromLeftEdge(PosX + CVarGetInteger("gMinimapPosX", 0));
+                                } else if (CVarGetInteger("gMinimapPosType", 0) == 2) {//Anchor Right
+                                    if (CVarGetInteger("gMinimapUseMargins", 0) != 0) {X_Margins_Minimap = Right_MM_Margin;};
+                                    entranceX = OTRGetRectDimensionFromRightEdge(PosX + CVarGetInteger("gMinimapPosX", 0));
+                                } else if (CVarGetInteger("gMinimapPosType", 0) == 3) {//Anchor None
+                                    entranceX = PosX + CVarGetInteger("gMinimapPosX", 0);
+                                }
                             } else if (CVarGetInteger("gMinimapPosType", 0) == 4) {//Hidden
-                                TopLeftX = -9999 << 2;
-                                TopLeftW = -9999 + IconSize << 2;
+                                entranceX = -9999;
```

Comment:
does this need to be in an `else` and not happen when `topLeft0` is false? just trying to think through how to make this logic easier to read

Reply:
its an else so that it happens even for topLeft0
basically topLeft0 is the authentic behavior when the icon is the top left (i.e. fix dungeon icon is off), but for cosmetic editor minimaps hidden we want the icon to be icon to be hidden even if its topLeft0

Other the other minimap options (anchor left, anchor right), should not move the topLeft0 icon.

Reply:
> it happens even for topLeft0

with this logic it's *only* for `topLeft0 == true`, but based on your description it seems that's intended

Reply:
Ok I reworked `topLeft0` to be more clear and commented. Now it just "overrides" the values at the end before rendering to set them back to 0,0 (from the left edge) or hides if minimap is disabled/the fix cvar is applied.